src/plugins/intel_cpu/tests/functional/subgraph_tests/src/gather_weights_decompression.cpp

```diff
@@ -0,0 +1,193 @@
+                                                              {},
+                                                              true);
+        auto subtract = std::make_shared<ov::op::v1::Subtract>(weights_convert, zp_convert);
+        auto multiply = std::make_shared<ov::op::v1::Multiply>(subtract, scale_const);
+        auto last_node = std::make_shared<ov::op::v0::Convert>(multiply, ov::element::f32);
+        return last_node;
+    }
+
+    std::shared_ptr<ov::Model> initSubgraph(const ov::PartialShape& data_shape,
+                                            const ov::Shape& weights_shape,
+                                            const ov::element::Type data_precision) {
+        ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(ov::element::i64, data_shape)};
+        auto params_convert = std::make_shared<ov::op::v0::Convert>(params[0], ov::element::i32);
+        auto axis = ov::op::v0::Constant::create(element::i32, Shape{1}, {0});
+
+        const auto weights_subgraph = initDecompressionWeights(weights_shape,
+                                                               ov::element::u8);
+
+        auto gather = std::make_shared<ov::op::v8::Gather>(weights_subgraph, params_convert, axis);
+
+        auto matB = ov::op::v0::Constant::create(element::f32, Shape{weights_shape[1], 1}, {1});
+        auto matMul = std::make_shared<ov::op::v0::MatMul>(gather, matB, false, false);
+        return makeNgraphFunction(data_precision, params, matMul, "GatherWeightsDecompression");
+    }
+
+    void SetUp() override {
+        targetDevice = ov::test::utils::DEVICE_CPU;
+
+        InputAndWeigthsShapeParams shape_params;
+        ov::AnyMap additional_config;
+        fusingSpecificParams fusing_params;
+        bool should_fuse;
+
+        std::tie(shape_params, additional_config, fusing_params, should_fuse) = GetParam();
+
+        configuration.insert(additional_config.begin(), additional_config.end());
+        std::tie(postOpMgrPtr, fusedOps) = fusing_params;
+        init_input_shapes({shape_params.data_shape, {{}, {{shape_params.weights_shape}}}});
+
+        ElementType netType = ov::element::f32;
+        inType = outType = netType;
+
+        function = initSubgraph(inputDynamicShapes[0], shape_params.weights_shape, netType);
+    }
+
+    void check_results() {
+        bool weights_found = false;
+        for (const auto& n : compiledModel.get_runtime_model()->get_ordered_ops()) {
+            if (n->get_friendly_name() == "Compressed_weights") {
+                ASSERT_EQ(n->get_output_element_type(0), ov::element::u8);
+                weights_found = true;
+            }
+        }
+        ASSERT_TRUE(weights_found);
+
+        CheckNumberOfNodesWithType(compiledModel, "Convert", 1);
+        CheckNumberOfNodesWithType(compiledModel, "Subtract", 0);
+        CheckNumberOfNodesWithType(compiledModel, "Multiply", 0);
+        CheckNumberOfNodesWithType(compiledModel, "Subgraph", 0);
```

Comment:
we should add more explicit checks to ensure Gather is really doing decompression, for example, by checking it's input/weight type as u8 and output type as f32/bf16.